File: unmaterialtab/src/main/java/com/ruslankishai/unmaterialtab/DimensUtils.java
package com.ruslankishai.unmaterialtab;

import android.content.Context;
import android.util.DisplayMetrics;
import android.util.TypedValue;

/**
 * Created by Ruslan Kishai aka creageek on 1/2/2017.
 */

public class DimensUtils {

    public static int spToPx(Context context, int sps) {
        return Math.round(context.getResources().getDisplayMetrics().scaledDensity * sps);
    }

    public static int dpToPx(Context context, int dps) {
        return Math.round(context.getResources().getDisplayMetrics().density * dps);
    }

    public static float spToPx(Context context, float spValue) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, spValue, dm);
    }

    public static float dpToPx(Context context, float dpValue) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dpValue, dm);
    }

}

File: unmaterialtab/src/main/java/com/ruslankishai/unmaterialtab/DimensUtils.kt
package com.ruslankishai.unmaterialtab

import android.content.Context
import android.util.DisplayMetrics
import android.util.TypedValue

/**
* Created by Ruslan Kishai on 1/2/2017.
*/

object DimensUtils {

    fun spToPx(context: Context, sps: Int): Int {
        return Math.round(context.resources.displayMetrics.scaledDensity * sps)
    }

    fun dpToPx(context: Context, dps: Int): Int {
        return Math.round(context.resources.displayMetrics.density * dps)
    }

    fun spToPx(context: Context, spValue: Float): Float {
        val dm = context.resources.displayMetrics
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, spValue, dm)
    }

    fun dpToPx(context: Context, dpValue: Float): Float {
        val dm = context.resources.displayMetrics
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dpValue, dm)
    }

}
